server/controllers/contest.js

```diff
@@ -85,4 +85,21 @@ exports.getContests = asyncHandler(async (req, res, next) => {
     } catch (error) {
     return res.status(500).send({ error });
     }
+})
+
+// return A list of contests that belongs to the userId
+exports.getContestsByUserId = asyncHandler(async (req, res, next) =>{
+    const userId = await User.findOne({_id: req.params.userId}).select('-password')
```

Comment:
We can avoid this call to the database by accessing the logged in user's id using the request object (`req.user.id`)
Please make sure you have the `protect` middleware applied to route and modify route accordingly.